urlaa914a1de2_xu_hao_erilex_tgz-pJ8-erilex_esl_ESLJ8/src/erilex/codegen/Class.java

```diff
@@ -0,0 +1,191 @@
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+EriLex is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with EriLex; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package erilex.codegen;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.*;
+
+/**
+ *
+ * @author ertri
+ */
+public class Class {
+
+    public String pack = "erilex.generated";
+    public boolean abs;
+    public String name;
+    public Type[] superType;
+    public List<String> params = new ArrayList<String>();
+    public List<Method> methods = new ArrayList<Method>();
+    public List<Variable> fields = new ArrayList<Variable>();
+    public List<Class> classes = new ArrayList<Class>();
+    public static final String PRE = "/** Code generated by EriLex */\n";
+    public boolean inner = false;
+    public boolean isStatic = false;
+    public String[] imports;
+    public String[] staticImports;
+    public boolean inter = false;
+
+    public Type getType() {
+        Type[] tparams = new Type[params.size()];
+
+        int i = 0;
+        for(String p : params) {
+            tparams[i++] = Generator.cons(p);
+        }
+        return Generator.cons(name, tparams);
+    }
+
+    public Class(String pack, boolean abs, String name, Type... superType) {
+        this.pack = pack;
+        this.abs = abs;
+        this.name = name;
+        this.superType = superType;
+    }
+
+    public Class(String pack, boolean abs, String name) {
+        this.pack = pack;
+        this.abs = abs;
+        this.name = name;
+    }
+
+    public Class(String pack, String name) {
+        this.pack = pack;
+        this.name = name;
+    }
+    public Class(String name) {
+        this.inner = true;
+        this.name = name;
+    }
+    public Class(boolean abs, String name, Type... superType) {
+        this.inner = true;
+        this.abs = abs;
+        this.name = name;
+        this.superType = superType;
+    }
+    public Class(String pack, String name, Type... superType) {
+        this.pack = pack;
+        this.name = name;
+        this.superType = superType;
+    }
+    public Class(String name, Type... superType) {
+        this.inner = true;
+        this.name = name;
+        this.superType = superType;
+    }
+
+    public void writeToFile(String path) throws IOException {
+        String[] s = pack.split(";")[0].split("\\.");
+        for(String d : s) {
+            path += File.separator+d;
+        }
+        File dir = new File(path);
+        if(!dir.exists()) {
+            dir.mkdirs();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Confirm that `mkdirs` was successful. Handle any failures that happened. If you fail silently, then you might encounter errors that are harder to debug. See the following diffs on GitHub that make this code change: [diff1](https://github.com/qunarcorp/qmq/commit/12f1d9d6a57a02bc2415cb2af9f4388a77b12c02#diff-6b67be6e17f24fdeddb7c725ac8ec5600d6e63be0bdca0963a83d530e6d55e35L36), [diff2](https://github.com/apache/jmeter/commit/1ec61206779b3f84df18516682b4791bf015178f#diff-cd74304763c097fbc085ac2bc952b46ab3843ac825cacacad89e8064296e955aL330)

Learn more: [File.mkdirs in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdirs--)